packages/compass-import-export/src/modules/export.ts

```diff
@@ -0,0 +1,537 @@
+  type: ExportActionTypes.ToggleFieldToExport,
+  fieldId,
+});
+
+type ToggleExportAllSelectedFieldsAction = {
+  type: ExportActionTypes.ToggleExportAllSelectedFields;
+};
+export const toggleExportAllSelectedFields = () => ({
+  type: ExportActionTypes.ToggleExportAllSelectedFields,
+});
+
+type AddFieldToExportAction = {
+  type: ExportActionTypes.AddFieldToExport;
+  path: SchemaPath;
+};
+export const addFieldToExport = (path: SchemaPath) => ({
+  type: ExportActionTypes.AddFieldToExport,
+  path,
+});
+
+type ReadyToExportAction = {
+  type: ExportActionTypes.ReadyToExport;
+  selectedFieldOption?: 'all-fields';
+};
+
+export const readyToExport = (): ReadyToExportAction => ({
+  type: ExportActionTypes.ReadyToExport,
+});
+
+type RunExportAction = {
+  type: ExportActionTypes.RunExport;
+};
+
+export const selectFieldsToExport = (): ExportThunkAction<
+  Promise<void>,
+  | SelectFieldsToExportAction
+  | FetchFieldsToExportAction
+  | FetchFieldsToExportErrorAction
+  | FetchFieldsToExportSuccessAction
+> => {
+  return async (dispatch, getState) => {
+    dispatch({
+      type: ExportActionTypes.SelectFieldsToExport,
+    });
+
+    const fieldsToExportAbortController = new AbortController();
+
+    dispatch({
+      type: ExportActionTypes.FetchFieldsToExport,
+      fieldsToExportAbortController,
+    });
+
+    const {
+      export: { query, namespace },
+      dataService: { dataService },
+    } = getState();
+
+    try {
+      const gatherFieldsResult = await gatherFieldsFromQuery({
+        ns: namespace,
+        abortSignal: fieldsToExportAbortController.signal,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        dataService: dataService!,
+        query,
+        sampleSize: 50,
+      });
+
+      if (
+        fieldsToExportAbortController.signal.aborted ||
+        gatherFieldsResult.aborted
+      ) {
+        dispatch({
+          type: ExportActionTypes.FetchFieldsToExportError,
+          aborted: true,
+        });
+        return;
+      }
+
+      const fields: FieldsToExport = {};
+      for (const schemaPath of gatherFieldsResult.paths) {
+        fields[getIdForSchemaPath(schemaPath)] = {
+          path: schemaPath,
+          // We start all of the fields as unchecked.
+          selected: false,
+        };
+      }
+
+      dispatch({
+        type: ExportActionTypes.FetchFieldsToExportSuccess,
+        fieldsToExport: fields,
+      });
+    } catch (err: any) {
+      if (fieldsToExportAbortController.signal.aborted) {
+        dispatch({
+          type: ExportActionTypes.FetchFieldsToExportError,
+          aborted: true,
+        });
+        return;
+      }
+
+      dispatch({
+        type: ExportActionTypes.FetchFieldsToExportError,
+        errorMessage: err?.message,
+      });
+    }
+  };
+};
+
+export const runExport = (): ExportThunkAction<
+  Promise<void>,
+  RunExportAction
+> => {
+  return async (dispatch) => {
+    // TODO(COMPASS-6580): Run export.
```

Comment:
So at this point export does not happen yet?

Reply:
Aah I just came looking for this :)

Reply:
Not yet no, this is feature flagged. It will be done in https://jira.mongodb.org/browse/COMPASS-6580